File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.py
import math

def check(num):
    check_arr = []
    if num == 1: return 0

    for i in range(2, int(math.sqrt(num)) + 1):
        if num % i == 0:
            check_arr.append(i)
            if num // i <= 10**7: return num // i
    
    if len(check_arr) != 0: return check_arr[-1]
    return 1

def solution(begin, end):
    answer = []
    
    for i in range(begin, end + 1):
        check_num = check(i)
        if check_num is not None: answer.append(check_num)
    
    return answer

File: 프로그래머스/2/12923. 숫자 블록/숫자 블록.js
function check(num) {       
    let checkArr = [];        
    if (num === 1) return 0;

    for (let i = 2; i <= Math.sqrt(num); i++){
        if (num % i === 0) {
            checkArr.push(i);
            if (num/i <= 10**7) return num/i;
        }
    }
    if (checkArr.length !== 0) return checkArr[checkArr.length-1];
    return 1;
}

function solution(begin, end) {
    let answer = [];
    
    for (let i = begin; i <= end; i++){
        let checkNum = check(i)
        if (checkNum !== undefined) answer.push(checkNum)
    }
    
    return answer;
}